privaterelay/signals.py

```diff
@@ -14,6 +21,36 @@ def record_user_signed_up(request, user, **kwargs):
     request.session.modified = True
 
 
+def _ses_message_props(data: str) -> ContentTypeDef:
+    return {"Charset": "UTF-8", "Data": data}
+
+
+@receiver(user_signed_up)
+def send_first_email(request, user, **kwargs):
+    welcome_email_flag = Flag.objects.filter(name="welcome_email").first()
+    has_welcome_email = welcome_email_flag and (
+        welcome_email_flag.everyone or welcome_email_flag.is_active_for_user(user)
+    )
+    if not has_welcome_email:
```

Comment:
Because this is in a request, you can add

```python
from waffle import flag_is_active
```

and then use:
```suggestion
    if not flag_is_active(request, 'welcome_email'):
```

There's no point in assigning this flag to a user - after they exist, they won't emit a `user_signed_up` signal, and you can't assign a flag to a user that doesn't exist. Waffle has a `switch` that would be more appropriate, but we haven't used switches before, so a flag with everyone on or off works fine.

Reply:
oh, nice! I was copying code from elsewhere, but I forgot this is all in a request context.